instrumentation/github.com/gin-gonic/gin/otelgin/gintrace.go

```diff
@@ -95,7 +95,9 @@ func Middleware(service string, opts ...Option) gin.HandlerFunc {
 			span.SetAttributes(semconv.HTTPStatusCode(status))
 		}
 		if len(c.Errors) > 0 {
-			span.SetAttributes(attribute.String("gin.errors", c.Errors.String()))
+			for _, err := range c.Errors {
+				span.RecordError(err.Err)
+			}
```

Comment:
(Sorry, I know this has been approved already)
Should we also set the span status to error?

Reply:
I saw that the setting was here, so I didn't make any settings.

Reply:
Additionally to be supplemented: Even if approved, as long as there is a need for adjustment, it can be pointed out. Try to make this component perfect as much as possible.

Reply:
> However, there are some differences here. In the existing logic, the status is set based on the status code. But in fact, it can also be explained through Error. So there are actually some differences here.

I can try to find some information based on this, look at best practices, see how Status is determined, and then consider how to adjust.

Reply:
https://opentelemetry.io/docs/concepts/signals/traces/#span-status

> To reiterate: Unset represents a span that completed without an error. Ok represents when a developer explicitly marks a span as successful. In most cases, it is not necessary to explicitly mark a span as Ok.

I saw this sentence on the official website, and my understanding is:

As long as we are very sure that there is no problem with him, it can mean OK; as long as there is an error (which may include status codes or may include error), we can set it to ERROR.

Does my understanding make sense?

Reply:
Could gin have a success HTTP status, but include errors?

Reply:
Then, we should mark the span as error if the HTTP status is an error, or if otelgin provides errors.

Reply:
Another supplementary material: https://opentelemetry.io/docs/specs/otel/trace/api/#set-status

This is its priority: Ok > Error > Unset

However, I checked the logic of the status codes, and it will only return Error and Unset. Therefore, it will be much easier for me to handle.

Reply:
Ok is a very specific status that should be set extremely rarely. We can keep Unset.

Reply:
Adjusted